packages/foreman/foreman/foreman.provreq

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python2
+
+"""
+Automatic Provides and Requires generator for Foreman's webpack bundles.
+
+Foreman ships webpack bundles in its RPMs to be consumed by plugins. To ensure
+proper dependency relationships between the plugins and the main Foreman RPM,
+this script generates Provides for the Foreman RPM that contain the hash of
+the webpack bundles shipped and matching Requires for the plugins packages.
+Each time the Foreman bundles change hashes, the plugins become uninstallable
+and indicate that a rebuild is required.
+"""
+
+import json
+import os
+import sys
+
+FOREMAN_MANIFEST = '/var/lib/foreman/public/webpack/manifest.json'
+
+
+def foreman_webpack(name, chunk):
+    filename, fileext = os.path.splitext(chunk)
+    fileext = fileext.lstrip('.')
+    filehash = filename.rsplit('-', 1)[1]
+    return 'foreman-webpack-{}-{}({})'.format(name, fileext, filehash)
+
+
+def parse_manifest(manifest_path):
+    webpack_chunks = set()
+
+    with open(manifest_path) as fh:
+        manifest = json.load(fh)
+
+    for name, content in manifest['assetsByChunkName'].items():
+        if name != 'vendor':
+            continue
+        if isinstance(content, basestring):
```

Comment:
This and `print` as a statement are the reason I initially made this python2 only. They're easy to work around:
```python
from __future__ import print_function
try:
    basestring
except NameError
    basestring = string
```

Reply:
done.